sklearn/manifold/t_sne.py

```diff
@@ -609,6 +573,18 @@ class TSNE(BaseEstimator):
         in the range of 0.2 - 0.8. Angle less than 0.2 has quickly increasing
         computation time and angle greater 0.8 has quickly increasing error.
 
+    n_jobs : integer (default: 1)
+        Only used if method='barnes_hut'
+        Number of CPU used to compute the nearest neighbors of each point.
+        If ``n_jobs=-1``, use all the CPUs.
+
+    neighbors_method : string or NeighborsBase object (default: 'ball_tree')
+        Only used if method='barnes_hut'
+        Method used to compute k nearest neighbors for Barnes-Hut T-SNE.
+        If it is a string, it should be compatible with the algorithm parameter
+        of `NearestNeighbors`. If it is an object, it should implement a
+        `kneighbors` method returning distances_nn and neighbors_nn.
```

Comment:
Actually this new parameter is a new feature that impacts the public API. In order to keep this PR focused on bug fixes, I think we should extract that specific change into a separate PR so as to be able to merge the bugfixes 0.19 and reschedule this improvement for 0.20.

We need a consistent naming convention for all the models that accept precomputed or pluggable nearest neighbors methods (e.g. DBSCAN is another example that comes to mind).